clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -256,11 +256,13 @@ public void poll(long timeout, long now, PollCondition pollCondition) {
      * nor will it execute any delayed tasks.
      */
     public void pollNoWakeup() {
-        disableWakeups();
-        try {
-            poll(0, time.milliseconds(), null);
-        } finally {
-            enableWakeups();
+        synchronized (this) {
```

Comment:
@hachikuji isn't the issue with that plan in `firePendingCompletedRequests`? We would have to hand that parameter down to the method since it also calls `poll`. Won't we change behavior of specifically `org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient#failUnsentRequests` to which we cannot hand down such a parameter, but which would have side effects from my suggestion [here](https://github.com/apache/kafka/pull/2699#discussion_r107456598) (and also the current solution)? Then again, maybe that change in behavior wouldn't be a bad thing?